### PR TITLE
EOS-281 SDK-453 [Node.js SDK] Add support for enhanced search

### PR DESCRIPTION
**Pull request by igor-alekseev-dev**
Pull Request Checklist
======================

Please read this for our PR Culture:
------------------------------------
https://incountry.atlassian.net/wiki/spaces/ED/pages/46858869/Pull+requests+culture+Draft

Work relates to:
----------------
Link(s) to Jira Task/User Story/Bug/Epic:
- [SDK-453](https://incountry.atlassian.net/browse/SDK-453)

Code Quality (Before submitting PR)
------------
- [x] Ensure code compiles and passes Sonar Lint.
- [x] Do your best to ensure integration tests pass. If they don't then...
- [x] Remove, ignore, or mark pending any brittle tests.
- [x] Remove commented out code.

Target Release
--------------
- [ ] Please put the target release vehicle this needs to be bundled with.
- [ ] Did you increment the version in package.json? What is the new version?

### CODE REVIEW

**Review by RommeroInCountry — CHANGES_REQUESTED** [review]
Let's bump version to 4.3.0 and add info on $like and $or to README. Codewise the PR is LGTM

File: README.md [hunk 1]
```diff
@@ -693,6 +712,14 @@ search (similar to the `LIKE` SQL operator, without special characters) within r
 { searchKeys: 'abc', key1: 'def' }
 ```
 
+Another way to find records by partial key match is using `$like` operator. It provides a partial match search (similar to the `LIKE` SQL operator without special characters) against one of the record’s string fields `key1, key2, ..., key20`.
+```typescript
+// Matches all records where record.key3 LIKE 'abc'
+{ key3: { $like: 'abc' } }
+```
+
+Note: You can use either `searchKeys` or `$like`.
```
Comment (RommeroInCountry) [inline README.md:721]: Note: You can use either `searchKeys` or `$like`, not both

File: README.md [hunk 2]
```diff
@@ -674,6 +675,24 @@ The following exact match search criteria are available:
 { rangeKey1: { $gte: 5, $lte: 100 } }
 ```
 
+- multiple criterion with `$or` operator:
```
Comment (RommeroInCountry) [inline README.md:678]: multiple criteria

**Review by RommeroInCountry — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -674,6 +675,24 @@ The following exact match search criteria are available:
...
+```typescript
+{
+  $or: [
+    { key1: 'john': key2: 'smith' },
```
Comment (artemgolozubov) [inline README.md:682]:
```suggestion
    { key1: 'john', key2: 'smith' },
```

**Review by artemgolozubov — COMMENTED** [review]
Please fix typos, the rest LGTM

File: README.md [hunk 4]
```diff
@@ -674,6 +675,24 @@ The following exact match search criteria are available:
...
+}
+
+{
+  $or: [
+    { key1: { $like: 'john' } },
+    { key2: { $like: 'john' } },
+  ]
+}
+```
+
```
Comment (RommeroInCountry) [inline README.md:694]:
```suggestion
Note: currently, nested `$or` operators are not supported
```

**Review by RommeroInCountry — CHANGES_REQUESTED** [review]

**Review by RommeroInCountry — APPROVED** [review]

**Review by artemgolozubov — APPROVED** [review]
LGTM

**Review by RommeroInCountry — APPROVED** [review]

**Pull request merged**